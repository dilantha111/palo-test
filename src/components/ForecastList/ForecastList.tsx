import { Box, List, ListItemButton, ListItemText } from "@mui/material";
import { ForeCastItem } from "../../types/ForeCastItem.type";
import React from "react";

export type Props = {
    forecastItems: ForeCastItem[],
    onClick: (forecastItem: ForeCastItem) => void
};

export default function ForecastList({ forecastItems, onClick }: Props) {
    const [selectedItem, setSelectedItem] = React.useState<ForeCastItem>();

    const handleListItemClick = (
        _event: React.MouseEvent<HTMLDivElement, MouseEvent>,
        forecastItem: ForeCastItem,
    ) => {
        setSelectedItem(forecastItem);
        onClick(forecastItem);
    };

    return <Box sx={{
        width: '100%', bgcolor: 'background.paper',
        maxHeight: '400px', overflow: 'scroll', borderColor: 'black', borderWidth: '1px'
    }}>
        <List component="nav" aria-label="main mailbox folders">
            {forecastItems.map(forecastItem => (
                <ListItemButton
                    key={forecastItem.locationName}
                    selected={selectedItem?.locationName === forecastItem.locationName}
                    onClick={(event) => handleListItemClick(event, forecastItem)}
                >
                    <ListItemText primary={forecastItem.locationName} />
                </ListItemButton>
            ))}
        </List>
    </Box>;
}