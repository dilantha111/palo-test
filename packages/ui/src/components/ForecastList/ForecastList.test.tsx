import { describe, it, expect, vi } from 'vitest';
import { render, fireEvent } from "@testing-library/react";
import ForecastList, { Props } from "./ForecastList";
import { ForeCastItem } from "../../types/ForeCastItem.type";

describe("ForecastList component", () => {
    const mockOnClick = vi.fn();

    const forecastItems: ForeCastItem[] = [
        {
            locationName: "Location 1",
            screenShotUrl: "https://location1.com",
            whetherForecast: "Sunny",
        },
        {
            locationName: "Location 2",
            screenShotUrl: "https://location2.com",
            whetherForecast: "Rainy",
        },
    ];

    const defaultProps: Props = {
        forecastItems: forecastItems,
        onClick: mockOnClick,
    };

    it("should render correctly", () => {
        const { container } = render(<ForecastList {...defaultProps} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it("should select the clicked item and call onClick", () => {
        const { getByText } = render(<ForecastList {...defaultProps} />);
        const item2 = getByText("Location 2");

        fireEvent.click(item2);

        expect(item2.parentNode?.parentNode).toHaveClass("Mui-selected");
        expect(mockOnClick).toHaveBeenCalledWith(forecastItems[1]);
    });
});
