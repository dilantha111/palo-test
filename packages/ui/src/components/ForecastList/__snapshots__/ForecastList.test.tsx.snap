// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ForecastList component > should render correctly 1`] = `
<div
  class="MuiBox-root css-5mbckt"
>
  <nav
    aria-label="main mailbox folders"
    class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  >
    <div
      class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          Location 1
        </span>
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          Location 2
        </span>
      </div>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </div>
  </nav>
</div>
`;
